File: 409-longest-palindrome/longest-palindrome.js
/**
 * @param {string} s
 * @return {number}
 */
var longestPalindrome = function(s) {
    if (s.length === 1){
        return 1;
    }
    let letterOccurrence = {};
    for (let letter of s){
        if (letter in letterOccurrence){
            letterOccurrence[letter]++;
        }
        else{
            letterOccurrence[letter] = 1;
        }
    }
    let palindrome = 0;
    for (key of Object.keys(letterOccurrence)){
        if (letterOccurrence[key] > 1){
            if (letterOccurrence[key] % 2 === 0){
                palindrome += letterOccurrence[key]
            }
            else{
                palindrome += letterOccurrence[key] - 1;
            }
        }
    }
    if (s.length > palindrome){
        palindrome++;
    }
    return palindrome;
};

File: 409-longest-palindrome/longest-palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        count = {}

        for letter in s:
            if count.get(letter, None):
                count[letter] += 1
            else:
                count[letter] = 1
        
        length_of_palindrome = 0
        for value in count.values():
            if value % 2 == 0:
                length_of_palindrome += value
            else:
                length_of_palindrome += value - 1
        return length_of_palindrome if len(s) == length_of_palindrome else length_of_palindrome + 1